src/Types/Animations/DownwardGravity.tsx

```diff
@@ -1,40 +1,55 @@
-import { SketchData } from '../Figures';
-import { AnimatedFigure } from '../ProcessingFigures';
-import { WallBounce } from './WallBounce';
+import { CustomFigureStyles, SketchData } from '../Figures';
+import p5 from 'p5';
+import { pushNewFigure } from './Animation';
 
 export class DownwardGravity extends Animation {
-    static draw(sketchData: SketchData, p) {
-        p.background(sketchData.colorSettings.background);
-
-        sketchData.figs.forEach(fig => {
-          if((fig.collideCanvasLeft(sketchData.canvasWidth, sketchData.canvasHeight) ||
-             fig.collideCanvasRight(sketchData.canvasWidth, sketchData.canvasHeight)) &&
-             !fig.dead) {
-            fig.velocity.x *= -1;
-            fig.spin *= -1;
-          }
-
-          let nodes = sketchData.figs;
-          
-          if (!fig.collideCanvasBottom(sketchData.canvasWidth, sketchData.canvasHeight)) {
-            fig.velocity.add(0, 0.07);
-            fig.pos.add(fig.velocity);
-            fig.angle += fig.spin;
-          }
-          else if(!fig.dead){
-            fig.dead = true;
-            fig.thud.play();
-          }
-
-          fig.display();
-        });
-    }
+  static draw(sketchData: SketchData, p: p5) {
+    let customStyles : CustomFigureStyles = {
+      opacity: 200,
+      stroke: false
+    };
+    
+    let color = p.color(sketchData.colorSettings.background);
+    color.setAlpha(30);
+    p.background(color);
 
-    static mousePressed(sketchData: SketchData, p) {
-        
+    if (p.mouseX != p.pmouseX && p.mouseY != p.pmouseY && Math.round(Math.random() * 2) === 1) {
+      pushNewFigure(sketchData.selectedFigure, sketchData.figs, p);
     }
 
-    static mouseReleased(sketchData: SketchData, p) {
+    sketchData.figs.forEach(fig => {
+
+      fig.pos.add(fig.velocity);
```

Comment:
When DownwardGravity is selected, but no shape is selected this line causes: `TypeError: Cannot read property 'pos' of undefined`

Reply:
Interesting, I'll look into it.

Reply:
Fixed